server/src/main/java/io/crate/geo/GeoJSONUtils.java

```diff
@@ -111,12 +111,13 @@ private static void forEach(Object arrayOrCollection, Consumer<Object> consumer)
     }
 
     public static Map<String, Object> shape2Map(Shape shape) {
-        if (shape instanceof ShapeCollection) {
-            ShapeCollection<?> shapeCollection = (ShapeCollection<?>) shape;
+        if (shape instanceof ShapeCollection<?> shapeCollection) {
             List<Map<String, Object>> geometries = new ArrayList<>(shapeCollection.size());
             for (Shape collShape : shapeCollection) {
                 geometries.add(shape2Map(collShape));
             }
+            // Even though we normalize input to transform Geometry Collection to MultiPoint or MultiPolygon,
+            // We don't need to do it here as this function operates with already normalized shapes.
```

Comment:
This should better move into a method doc.  Is this related at all to the issue? Why should we normalize shapes?

Reply:
We shouldn't normalize shapes, I had a concern that we can de-normalize a valid Multi* shape and turn it to a `GeometryCollection` but I couldn't find a real scenario reflecting this.

Test with smth like `compare(..., type.implicitCast(GeoJSONUtils.map2Shape(geoJson))`) used to fail without sanitizing the result here but as it's not real scenario, I will remove this comment